charms/kfp-persistence/src/components/pebble_components.py

```diff
@@ -0,0 +1,95 @@
+    """Defines configuration for PersistenceAgent Service."""
+
+    KFP_API_SERVICE_NAME: str
+    NAMESPACE: str
+
+
+class PersistenceAgentPebbleService(PebbleServiceComponent):
+    """Pebble Service for Persistence Agent Container."""
+
+    def __init__(
+        self,
+        *args,
+        environment: Dict[str, str],
+        **kwargs,
+    ):
+        """Initialize component."""
+        super().__init__(*args, **kwargs)
+        self._environment = environment
+
+    def get_layer(self) -> Layer:
+        """Pebble configuration layer for persistenceagent.
+
+        This method is required for subclassing PebbleServiceComponent
+        """
+        logger.info("PersistenceAgentContainer: create layer")
+
+        # retrieve up-to-date service configuration as setup by charm
+        try:
+            service_config: PesistenceAgentServiceConfig = self._inputs_getter()
+        except Exception as err:
+            logger.error(f"PersistenceAgentContainer: configuration is not provided: {err}")
+            return None
+
+        if len(service_config.KFP_API_SERVICE_NAME) == 0 or len(service_config.NAMESPACE) == 0:
+            logger.info("PersistenceAgentContainer: configuration is not valid")
+            return None
+
+        # setup command with parameters provided in configuration
+        command = (
+            "persistence_agent",
+            "--logtostderr=true",
+            f"--namespace={service_config.NAMESPACE}",
+            "--ttlSecondsAfterWorkflowFinish=86400",
+            "--numWorker=2",
+            f"--mlPipelineAPIServerName={service_config.KFP_API_SERVICE_NAME}",
+        )
+
+        # generate and return layer
+        return Layer(
+            {
+                "services": {
+                    self.service_name: {
+                        "override": "replace",
+                        "summary": "persistenceagent service",
+                        "command": " ".join(command),
+                        "startup": "enabled",
+                        "environment": self._environment,
+                    }
+                },
+                "checks": {
+                    "persistenceagent-get": {
+                        "override": "replace",
+                        "period": "30s",
+                        "http": {"url": "http://localhost:8080/metrics"},
+                    }
+                },
+            }
+        )
+
+    def get_status(self) -> StatusBase:
+        """Return status."""
+        # validate configuration availability
+        try:
+            service_config: PesistenceAgentServiceConfig = self._inputs_getter()
+        except Exception as err:
+            return WaitingStatus(f"Configuration is not provided: {err}")
+
+        # validate values
+        if len(service_config.KFP_API_SERVICE_NAME) == 0 or len(service_config.NAMESPACE) == 0:
+            return WaitingStatus("Configuration is not valid")
+
+        return ActiveStatus()
```

Comment:
The parent class already has a `get_status` that checks for container/service readiness - I think we need to extend that rather than replace.  Maybe:

```suggestion
    def get_status(self) -> StatusBase:
        """Return status."""
        # validate configuration availability
        try:
            service_config: PesistenceAgentServiceConfig = self._inputs_getter()
        except Exception as err:
            return WaitingStatus(f"Configuration is not provided: {err}")

        # validate values
        if len(service_config.KFP_API_SERVICE_NAME) == 0 or len(service_config.NAMESPACE) == 0:
            return WaitingStatus("Configuration is not valid")

        return super().get_status()
```
(I think that is the right syntax?)

Reply:
Will do.